server/src/main/java/io/druid/client/coordinator/CoordinatorClient.java

```diff
@@ -61,38 +62,44 @@ public CoordinatorClient(
     this.selector = selector;
   }
 
+  public List<String> findDatasources(List<String> dataSources)
```

Comment:
Ah hmm. Looks like this is being used to do an HTTP call to the coordinator to resolve the datasource names. That shouldn't be necessary, since the broker already has a cache with enough info (the server inventory view it uses to compute `/druid/v2/datasources` in `ClientInfoResource`). It'd be better to use that instead.